src/App.tsx

```diff
@@ -1,26 +1,221 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useMemo, useRef, useState } from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+// import { getTodos, USER_ID } from './api/todos';
+import { Todo } from './types/Todo';
+import { Header } from './components/Header';
+import { Footer } from './components/Footer';
+import { Status } from './types/Status';
+import { TodoList } from './components/TodoList';
+import { Notification } from './components/Notification';
+import * as todoService from './api/todos';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [filterStatus, setFilterStatus] = useState<Status>(Status.All);
+  const [newTitle, setNewTitle] = useState('');
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [loading, setLoading] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+
+  const visibleTodos = useMemo(() => {
+    return todos.filter(todo => {
+      switch (filterStatus) {
+        case Status.All:
+          return true;
+
+        case Status.Active:
+          return !todo.completed;
+
+        case Status.Completed:
+          return todo.completed;
+      }
+    });
+  }, [todos, filterStatus]);
+
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (inputRef.current) {
+      inputRef.current.focus();
+    }
+  }, [loading]);
+
+  const errorTimeout = useRef<number | null>(null);
+
+  const showError = (message: string) => {
+    setErrorMessage(message);
+
+    if (errorTimeout.current) {
+      clearTimeout(errorTimeout.current);
+    }
+
+    errorTimeout.current = window.setTimeout(() => {
+      setErrorMessage('');
+    }, 3000);
+  };
+
+  function loadTodos() {
+    setLoading(true);
+
+    todoService
+      .getTodos()
+      .then(setTodos)
+      .catch(() => {
+        showError('Unable to load todos');
+      })
+      .finally(() => setLoading(false));
+  }
+
+  useEffect(() => {
+    loadTodos();
+  }, [todoService.USER_ID]);
+
+  if (!todoService.USER_ID) {
     return <UserWarning />;
   }
 
+  const completedTodos = todos.filter(todo => todo.completed);
+  const activeTodos = todos.filter(todo => !todo.completed);
+  const allTodosAreActive =
+    todos.length > 0 && todos.every(todo => todo.completed);
+
+  function addTodo(newTodo: Omit<Todo, 'id'>) {
```

Comment:
It is important to follow the same way of creating functions everywhere, fix it
```suggestion
  const addTodo = (newTodo: Omit<Todo, 'id'>) => {
```